dashboard/app/models/user.rb

```diff
@@ -157,6 +157,12 @@ class User < ActiveRecord::Base
   accepts_nested_attributes_for :school_info, reject_if: :preprocess_school_info
   validates_presence_of :school_info, unless: :school_info_optional?
 
+  validate :must_be_a_teacher_if_has_sections, on: :update, if: -> {user_type_changed? && student?}
```

Comment:
I just checked on production-console, and here are the numbers:
 - invalid (with current rules): 166
 - no user: 4665
 - student user: 1377

All 166 currently invalid ones are missing [name](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/section.rb#L39). The last of those was created in 2014, however the latest follower was added as recently as December 2016.

The 1377 sections with a student user are what would raise the honeybadger error that prompted this.

The remaining 4665 have missing (soft-deleted presumably) teachers. These fail validity checks, for instance:
```
irb(main):078:0> Section.find(52).valid?
NoMethodError: undefined method `user_type' for nil:NilClass
```
but it appears they can still be joined. You're right, invalid sections abound.

So how about for now, I'll scope this down to fixing the nil bug in the section validation and adding a check for `user && user.teacher?` in `follower_controller`?